torchao/prototype/float8nocompile/kernels/fp8_dynamic_tensorwise_test.py

```diff
@@ -2,62 +2,153 @@
 import torch
 from torchao.float8.float8_scaling_utils import hp_tensor_to_float8_dynamic
 from torchao.float8.float8_tensor import LinearMMConfig
+from torchao.float8.float8_utils import is_row_major
 from torchao.prototype.float8nocompile.kernels.fp8_dynamic_tensorwise import (
+    hp_to_fp8_col_major,
+    hp_to_fp8_row_major,
     KernelAlgorithm,
-    triton_hp_tensor_to_float8_dynamic,
 )
 
 
 @pytest.mark.parametrize(
-    "algo", [KernelAlgorithm.ATOMIC_MAX, KernelAlgorithm.REDUCTION]
+    "algo",
+    [KernelAlgorithm.REDUCTION, KernelAlgorithm.ATOMIC_MAX],
 )
 @pytest.mark.parametrize(
     "input_shape",
-    [(32, 32), (512, 512), (4096, 4096)],
+    [(2, 4), (32, 16), (512, 512)],
 )
-def test_fp8_triton_hp_tensor_to_float8_dynamic(
-    algo: KernelAlgorithm, input_shape: tuple[int, int]
-):
+def test_fp8_hp_to_fp8_row_major(input_shape: tuple[int, int], algo: KernelAlgorithm):
     assert torch.cuda.is_available()
     device = "cuda"
     input_bf16 = torch.randn(input_shape, dtype=torch.bfloat16, device=device)
     x_bf16 = input_bf16.clone().detach().to(device)
     y_bf16 = input_bf16.clone().detach().to(device)
 
     # production implementation
-    x_fp8 = hp_tensor_to_float8_dynamic(
+    x_fp8_row_major = hp_tensor_to_float8_dynamic(
         x_bf16,
         torch.float8_e4m3fn,
         LinearMMConfig(),
     )
 
     # float8nocompile triton implementation
-    y_fp8 = triton_hp_tensor_to_float8_dynamic(
+    y_fp8_row_major = hp_to_fp8_row_major(
         y_bf16,
         torch.float8_e4m3fn,
         LinearMMConfig(),
         algo=algo,
     )
 
-    def allclose_fp8(tensor1, tensor2, atol=1e-3, rtol=1e-3):
-        # convert fp8 tensors to a higher precision (e.g., float32) for comparison
-        # since torch.allclose does not support fp8 tensors
-        if tensor1.shape != tensor2.shape:
-            raise ValueError("Tensors must have the same shape for comparison.")
-        if tensor1.dtype != tensor2.dtype:
-            raise ValueError("Tensors must have the same dtype for comparison.")
+    # check scales
+    assert torch.allclose(
+        x_fp8_row_major._scale, y_fp8_row_major._scale, atol=1e-3, rtol=1e-3
```

Comment:
can this be an exact match?

Reply:
Done. Also using exact match allowed me to discover and fix a small bug (converting amax to fp32 instead of fp64 when calculating the scale). :)